src/components/tipRecords/AuthorAndDate.vue

```diff
@@ -1,48 +1,101 @@
 <template>
-  <div
-    class="author-and-date"
-    @click.stop
-  >
-    <RouterLink
-      :to="{
-        name: 'user-profile',
-        params: {
-          address,
-        },
-      }"
+  <div class="wrapper">
+    <UserHoverCard
+      :address="address"
+      :hover="hover"
+      :show="show"
+      :is-active="isActive"
+      :mouseover="mouseover"
+      :mouseleave="mouseleave"
+      :mousestay="mousestay"
+    />
+    <div
+      class="author-and-date"
+      @click.stop
     >
-      <Avatar :address="address" />
-      <div class="author-name">
-        <span
-          v-if="name"
-          class="chain-name"
-        >
-          {{ name }}
-        </span>
-        <span class="address">
-          {{ address }}
-        </span>
-      </div>
-    </RouterLink>
-    <span class="right">
-      <FormatDate v-bind="$attrs" />
-      <slot />
-    </span>
+      <RouterLink
+        :to="{
+          name: 'user-profile',
+          params: {
+            address,
+          },
+        }"
+      >
+        <Avatar
+          :address="address"
+          @mouseover.native="mouseover"
+          @mouseleave.native="mouseleave"
+        />
+        <div class="author-name">
+          <span
+            v-if="name"
+            class="chain-name"
+          >
+            {{ name }}
+          </span>
+          <span class="address">
+            {{ address }}
+          </span>
+        </div>
+      </RouterLink>
+      <span class="right">
+        <FormatDate v-bind="$attrs" />
+        <slot />
+      </span>
+    </div>
   </div>
 </template>
 
 <script>
+import { mapMutations, mapState } from 'vuex';
 import FormatDate from './FormatDate.vue';
 import Avatar from '../Avatar.vue';
+import UserHoverCard from '../UserHoverCard.vue';
+
+let timeout;
```

Comment:
I don't think that we need the same `timeout` variable for all components. Consider moving it into `data`.